python/graphstorm/gconstruct/transform.py

```diff
@@ -255,12 +282,152 @@ def process_labels(data, label_confs, is_node):
+        return res
+
+class RegressionProcessor(LabelProcessor):
+    """ Process the label for the regression task.
+    """
+
+    def __call__(self, data):
+        """ Process the label for regression.
+
+        This performs data split on the nodes/edges and generates training/validation/test masks.
+
+        Parameters
+        ----------
+        data : dict of Tensors
+            All data associated with nodes/edges of a node/edge type.
+
+        Returns
+        -------
+        dict of Tensors : it contains the labels as well as training/val/test splits.
+        """
+        assert self.col_name in data, f"The label column {self.col_name} does not exist."
+        label = data[self.col_name]
+        valid_label_idx = get_valid_label_index(label)
+        def permute_idx():
+            return np.random.permutation(valid_label_idx)
+        res = self.data_split(permute_idx, len(label))
+        res[self.label_name] = label
+        return res
+
+class LinkPredictionProcessor(LabelProcessor):
+    """ Process the label for the link prediction task.
+    """
+
+    def __call__(self, data):
+        """ Process the label for link prediction.
+
+        This performs data split on the edges and generates training/validation/test masks.
+
+        Parameters
+        ----------
+        data : dict of Tensors
+            All data associated with nodes/edges of a node/edge type.
+
+        Returns
+        -------
+        dict of Tensors : it contains training/val/test splits for link prediction.
+        """
+        # Any column in the data can define the number of samples in the data.
+        assert len(data) > 0, "The edge data is empty."
+        for val in data.values():
+            num_samples = len(val)
+            break
+        def permute_idx():
+            return np.random.permutation(num_samples)
+        return self.data_split(permute_idx, num_samples)
+
+def parse_label_ops(confs, is_node):
+    """ Parse the configurations to generate the label processor.
+
+    Parameters
+    ----------
+    confs : dict
+        Contain the configuration for labels.
+    is_node : bool
+        Whether the configurations are defined for nodes.
+
+    Returns
+    -------
+    list of LabelProcessor : the label processors generated from the configurations.
+    """
+    assert len(confs) == 1, "We only support one label per node/edge type."
+    label_conf = confs[0]
+    assert 'task_type' in label_conf, "'task_type' must be defined in the label field."
+    task_type = label_conf['task_type']
+    # By default, we use all labels for training.
+    split_pct = label_conf['split_pct'] if 'split_pct' in label_conf else [1, 0, 0]
+
+    if task_type == 'classification':
+        assert 'label_col' in label_conf, \
+                "'label_col' must be defined in the label field."
+        label_col = label_conf['label_col']
+        return [ClassificationProcessor(label_col, label_col, split_pct)]
```

Comment:
We need to think about the case when a user provides valid and test set.

Reply:
let's address this in another PR.